x/blob/module_simulation.go

```diff
@@ -0,0 +1,67 @@
+package blob
+
+import (
+	"math/rand"
+
+	"github.com/celestiaorg/celestia-app/testutil/sample"
+	bloblsimulation "github.com/celestiaorg/celestia-app/x/blob/simulation"
+	"github.com/celestiaorg/celestia-app/x/blob/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	simappparams "github.com/cosmos/cosmos-sdk/simapp/params"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/authz/codec"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+)
+
+// avoid unused import issue
+var (
+	_ = sample.AccAddress
+	_ = bloblsimulation.FindAccount
+	_ = simappparams.StakePerAccount
+	_ = simulation.MsgEntryKind
+	_ = baseapp.Paramspace
+)
+
+// GenerateGenesisState creates a randomized GenState of the module
+func (AppModule) GenerateGenesisState(simState *module.SimulationState) {
+	accs := make([]string, len(simState.Accounts))
+	for i, acc := range simState.Accounts {
+		accs[i] = acc.Address.String()
+	}
+	blobGenesis := types.GenesisState{
+		Params: types.DefaultParams(),
+	}
+	simState.GenState[types.ModuleName] = simState.Cdc.MustMarshalJSON(&blobGenesis)
+}
+
+// ProposalContents doesn't return any content functions for governance proposals
+func (AppModule) ProposalContents(_ module.SimulationState) []simtypes.WeightedProposalContent {
+	return nil
+}
+
+// RandomizedParams creates randomized  param changes for the simulator
+func (am AppModule) RandomizedParams(_ *rand.Rand) []simtypes.ParamChange {
+	blobParams := types.DefaultParams()
+	return []simtypes.ParamChange{
+		simulation.NewSimParamChange(types.ModuleName, string(types.KeyMinSquareSize), func(r *rand.Rand) string {
+			return string(codec.Amino.MustMarshalJSON(blobParams.MinSquareSize))
+		}),
+		simulation.NewSimParamChange(types.ModuleName, string(types.KeyMaxSquareSize), func(r *rand.Rand) string {
+			return string(codec.Amino.MustMarshalJSON(blobParams.MaxSquareSize))
+		}),
+		simulation.NewSimParamChange(types.ModuleName, string(types.KeyGasPerBlobByte), func(r *rand.Rand) string {
+			return string(codec.Amino.MustMarshalJSON(blobParams.GasPerBlobByte))
+		}),
+	}
+}
```

Comment:
[blocking question]
I'm guessing that this uses a random parameter when booting this up. How does know to pick a power of two for the square size? if its not picking a power of two, shouldn't this be failing?